cob_monitoring/src/ping_computers.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Mojin Robotics GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rospy
+import subprocess
+
+from std_msgs.msg import String
+from diagnostic_msgs.msg import DiagnosticStatus, KeyValue
+
+class ping_computers:
+    def __init__(self):
+        self.ping_number_packages = rospy.get_param('~ping_number_packages', None)
+
+        self.hosts = rospy.get_param('~host_ips', None)
+        if not type(self.hosts) == list:
+            rospy.logerr("Ping Computers: hosts are not a list: %s", str(self.hosts))
+            exit
+
+        self.host_pc = rospy.get_param('~host_pc', None)
+        self.hosts_pingable = {}
+        diagnostics_frequency = rospy.get_param('~diagostics_frequency', None)
+        if diagnostics_frequency < 0:
+            rospy.logerr("Ping Computers: invalid frequency: %f", diagnostics_frequency)
+            exit            
+        if self.host_pc == "ROBOT":
+            # on robot, publish diagnostics
+            self.diagnostics_publisher = rospy.Publisher('/diagnostics', DiagnosticStatus,queue_size=1)
+            rospy.Timer(rospy.Duration(1.0/diagnostics_frequency), self.publish_diagnostics)
+            # use all but last digits from local IP address
+            ip_prefix = ".".join(subprocess.check_output(['hostname', '-I']).split(".")[:-1])
+            rospy.logdebug("Ping Computers: computer IP: %s, IP prefix: %s",subprocess.check_output(['hostname', '-I']), ip_prefix)
+            for host in self.hosts:
+                host_ip = ip_prefix+"."+str(host)
+                self.hosts_pingable.update({host_ip:False})
+        elif self.host_pc == "ROBOWATCH":
+            # publish directly to topic when running on robowatch pc
+            publish_topic = rospy.get_param('~publish_topic', None)
+            self.slack_publisher = rospy.Publisher(publish_topic, String, queue_size=1)
+            rospy.Timer(rospy.Duration(1.0/diagnostics_frequency), self.publish_slack)
```

Comment:
no need for an extra timer...just call publish from `publish_diagnostics` - this also remove duplicate code in both functions

Reply:
only publish on error, then stop until ok again

Reply:
also, if we split the two variants, then "local" can publish diagnostics and "remote" would `zmq_push`